### PR TITLE
Add recipe for reportlab 3.6.9

### PR DESCRIPTION
**Pull request by ivanholmes**

### CODE REVIEW

**Comment by ivanholmes** [comment]
Just noticed I have submitted this without filling the description properly! I will sort it shortly.
Sorry for the noise

**Comment by ivanholmes** [comment]
Description now sorted

File: dev-python/reportlab/patches/reportlab-3.6.9.patchset [hunk 1]
```diff
@@ -0,0 +1,46 @@
...
+diff --git a/setup.py b/setup.py
+index 3b7e665..3ad2c8a 100644
+--- a/setup.py
++++ b/setup.py
+@@ -252,6 +252,9 @@ class inc_lib_dirs:
+             # darwin ports installation directories
+             aDir(L, "/opt/local/lib")
+             aDir(I, "/opt/local/include")
++        elif platform.startswith("haiku"):
++        	aDir(I, "/boot/system/develop/headers/freetype2")
```
[the remark this answers is not in the document]
Comment (extrowerk) [reply dev-python/reportlab/patches/reportlab-3.6.9.patchset:16]: Python3 is only available for secondary arch on x86, but as your include folder order is primary-arch then secondary-arch it is possible the compiler will find the wrong set of headers.
This is also true for the libs.

I would propose to flip the folder order.

[the remark this answers is not in the document]
Comment (ivanholmes) [reply dev-python/reportlab/patches/reportlab-3.6.9.patchset:17]: Done, thanks

File: dev-python/reportlab/reportlab-3.6.9.recipe [hunk 2]
```diff
@@ -0,0 +1,74 @@
...
+
+ARCHITECTURES="all !x86_gcc2"
+SECONDARY_ARCHITECTURES="x86"
+
+PROVIDES="
+	$portName = $portVersion
+	"
+REQUIRES="
+	haiku$secondaryArchSuffix
+	lib:libfreetype$secondaryArchSuffix
```
Comment (Begasus) [inline dev-python/reportlab/reportlab-3.6.9.recipe:27]:
```diff
- 	lib:libfreetype$secondaryArchSuffix
```

File: dev-python/reportlab/reportlab-3.6.9.recipe [hunk 3]
```diff
@@ -0,0 +1,74 @@
...
+PYTHON_PACKAGES=(python3 python38 python39 python310)
+PYTHON_VERSIONS=(3.7 3.8 3.9 3.10)
+for i in "${!PYTHON_PACKAGES[@]}"; do
+pythonPackage=${PYTHON_PACKAGES[i]}
+pythonVersion=${PYTHON_VERSIONS[$i]}
+eval "PROVIDES_${pythonPackage}=\"\
+	${portName}_$pythonPackage = $portVersion\
+	\"; \
+REQUIRES_$pythonPackage=\"\
+	haiku\n\
```
Comment (Begasus) [inline dev-python/reportlab/reportlab-3.6.9.recipe:48]:
```diff
+ 	lib:libfreetype$secondaryArchSuffix\n\
```

Comment (Begasus) [inline dev-python/reportlab/reportlab-3.6.9.recipe:48]:
```diff
+ 	lib:libfreetype$secondaryArchSuffix\n\
```

**Review by Begasus — CHANGES_REQUESTED** [review]
libfreetype is linked in and I think it's needed for the subpackages as a requirement

**Comment by ivanholmes** [comment]
I agree, I'm away from my Haiku machine for the weekend but will push this change upon my return


________________________________
De : Schrijvers Luc ***@***.***>
Envoyé : vendredi 15 avril 2022 10:47
À : haikuports/haikuports
Cc: Ivan Holmes; Author
Objet : Re: [haikuports/haikuports] Add recipe for reportlab 3.6.9 (PR #6825)

@Begasus requested changes on this pull request.

libfreetype is linked in and I think it's needed for the subpackages as a requirement

________________________________

In dev-python/reportlab/reportlab-3.6.9.recipe:

> +REVISION="1" 
+pypi="81ff7e3f9fd345cd4685f964fbc3d89a06f39a4f552ab1c2a5769a0f9013" 
+SOURCE_URI="https://files.pythonhosted.org/packages/16/31/$pypi/reportlab-$portVersion.tar.gz" 
+CHECKSUM_SHA256="5d0cc3682456ad213150f6dbffe7d47eab737d809e517c316103376be548fb84" 
+SOURCE_DIR="reportlab-$portVersion" 
+PATCHES="reportlab-$portVersion.patchset" 
+ 
+ARCHITECTURES="all !x86_gcc2" 
+SECONDARY_ARCHITECTURES="x86" 
+ 
+PROVIDES=" 
+ $portName = $portVersion 
+ " 
+REQUIRES=" 
+ haiku$secondaryArchSuffix 
+ lib:libfreetype$secondaryArchSuffix 

-  lib:libfreetype$secondaryArchSuffix

________________________________

In dev-python/reportlab/reportlab-3.6.9.recipe:

> + " 
+BUILD_PREREQUIRES=" 
+ cmd:gcc$secondaryArchSuffix 
+ cmd:git 
+ " 
+ 
+PYTHON_PACKAGES=(python3 python38 python39 python310) 
+PYTHON_VERSIONS=(3.7 3.8 3.9 3.10) 
+for i in "${!PYTHON_PACKAGES[@]}"; do 
+pythonPackage=${PYTHON_PACKAGES[i]} 
+pythonVersion=${PYTHON_VERSIONS[$i]} 
+eval "PROVIDES_${pythonPackage}=\"\ 
+ ${portName}_$pythonPackage = $portVersion\ 
+ \"; \ 
+REQUIRES_$pythonPackage=\"\ 
+ haiku\n\ 

+  lib:libfreetype$secondaryArchSuffix\n\

________________________________

In dev-python/reportlab/reportlab-3.6.9.recipe:

> + " 
+BUILD_PREREQUIRES=" 
+ cmd:gcc$secondaryArchSuffix 
+ cmd:git 
+ " 
+ 
+PYTHON_PACKAGES=(python3 python38 python39 python310) 
+PYTHON_VERSIONS=(3.7 3.8 3.9 3.10) 
+for i in "${!PYTHON_PACKAGES[@]}"; do 
+pythonPackage=${PYTHON_PACKAGES[i]} 
+pythonVersion=${PYTHON_VERSIONS[$i]} 
+eval "PROVIDES_${pythonPackage}=\"\ 
+ ${portName}_$pythonPackage = $portVersion\ 
+ \"; \ 
+REQUIRES_$pythonPackage=\"\ 
+ haiku\n\ 

+  lib:libfreetype$secondaryArchSuffix\n\

**Review by Begasus — APPROVED** [review]

**Comment by Begasus** [comment]
Thanks! Keep an eye on the buildmaster for this :)

**Comment by ivanholmes** [comment]
> Thanks! Keep an eye on the buildmaster for this :)

Brilliant thanks for your assistance

**Pull request merged**